data/src/message.rs

```diff
@@ -1139,12 +1325,21 @@ fn target(
     }
 }
 
-pub fn message_id(message: &Encoded) -> Option<String> {
+pub type MsgId = Arc<str>;
```

Comment:
I am not entirely sure i like `MsgId` as name here. Maybe `Id` is better since we are in message crate after all.
`message::MsgId` vs `message::Id`